src/ast/passes/codegen_llvm.cpp

```diff
@@ -2367,27 +2367,32 @@ void CodegenLLVM::visit(AssignMapStatement &assignment)
     b_.CreateLifetimeEnd(val);
 }
 
-void CodegenLLVM::visit(AssignVarStatement &assignment)
+void CodegenLLVM::updateVariableType(const std::string &var_ident,
```

Comment:
This function name's a bit confusing as it won't actually update the variable's type if the variable is already defined